src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -2257,7 +2257,8 @@ PyMODINIT_FUNC PyInit_lpython_module_)" + fn_name + R"((void) {
             }
             default: throw CodeGenError("BinOp: " + std::to_string(x.m_op) + " operator not implemented yet");
         }
-        src = "";
+        src = "(" + CUtils::get_c_type_from_ttype_t(x.m_type) + ")";
```

Comment:
For unsigned ints, we are needing to explicitly cast them back to unsigned int. (C seems to promote unsigned ints to signed integers which leads to unexpected errors).

Reply:
I would only do this when it is needed. In particular, since shift returns signed in C, I would cast it to unsigned, but only for shifts. Not for any other operation, such as Add.

Reply:
To get started, for unsigned integers just cast the result to unsigned. Later we can refine it more.

Reply:
Got it. Updated. Thank you for the guidance.